### ISSUE TITLE
Dark mode

### ISSUE THREAD

**Issue by irilyth** [buttonmen-dev/buttonmen#2563]
Many web sites support the idea of a "dark mode"; it'd be cool if ours did too. (PoshFrosh suggested this in in-game chat with me in http://www.buttonweavers.com/ui/game.html?game=59428)

**Comment by jl8e** [comment]
The obvious complication is that people may want different color settings for dark mode?

**Comment by blackshadowshade** [comment]
Am I right in just thinking that this is simply a change of the linked CSS file? Or is there more to it?

@AdmiralJota, feel like weighing in on this?

**Comment by jl8e** [comment]
Yes, or there may be a way to do it within a single file.

The CSS is loaded before any JS runs, though.

**Comment by blackshadowshade** [comment]
I've just checked this issue and noticed the update with the blog post. Doesn't look too difficult, but I'd still love @AdmiralJota to give me a heads up on his thoughts.

**Comment by danlangford** [comment]
have you guys considered using a small component library on the front end to give a modern, clean polished look? in addition to modern looks they also improve mobile layout and screen reader support. Bootstrap is definitely popular but i personally have enjoyed using Halfmoon. its smaller, lighter weight, and has built in dark mode support. i use it on my Button Filter if you want to check it out. http://bit.ly/ButtonFilter

**Comment by blackshadowshade** [comment]
I guess the question that I always have is how much work it is to do such a transition---and how would such a transition look in practice?

I've used Bootstrap a bit at work, and I know that it takes quite a bit of work to transition over, including redefining the look and feel of the page into its 12-column system.

**Comment by danlangford** [comment]
![image](https://user-images.githubusercontent.com/721364/101136448-96869380-35ca-11eb-928e-69e34549a2d2.png)

![image](https://user-images.githubusercontent.com/721364/101136481-a1412880-35ca-11eb-9a39-05425c5f4908.png)

this is just a quick thrown together HTML of the Open games screen. i rearranged a few things but not much. i think you could rearrange a lot more (put profile and preferences inside a dropdown attached to the username). this wasn't tuned at all. i didnt have to fight the 12 column layout at all. though i personally think the 12 column layout is really powerful to help you feel subconsciously that all is right in the world because everything lines up. its clear some things would need to be tuned but this was about 30-40 minutes of tinkering. so this is just an example.

**Comment by blackshadowshade** [comment]
Could you add a Gist for the code (or post it wholescale here) too? That'll give me some idea of what sort of changes we're looking at.

**Comment by danlangford** [comment]
here is a gist. it obviously has some work that needs to be done before you drop it in but its just an idea
https://gist.github.com/danlangford/866c001739a9eec6699f55d4df2724d3
my mock up didnt have the `darkmode` button because i wanted it to be fairly close to the existing site layout. if you run it locally just press `Shift+d` to toggle darkmode.

**Comment by danlangford** [comment]
And to be clear I would love to do some work that could be pushed out to a dev site for consideration. And maybe we should leave this thread about darkmode and conversations about deeper UI changes should be over on #1401 or a new issue
